### PR TITLE
fix: Improve device movement of scalers

### PR DESCRIPTION
**Pull request by HCookie**
## Description
Scalers were initialised as numpy arrays and unneccessary movement between devices was occuring. 
This simplifies the movement paradigm.

Part of https://github.com/ecmwf/anemoi-core/issues/370

By opening this pull request, I affirm that all authors agree to the [Contributor License Agreement.](https://github.com/ecmwf/codex/blob/main/Legal/contributor_license_agreement.md)

### CODE REVIEW

**Review by ssmmnn11 — APPROVED** [review]
looks good.

File: training/src/anemoi/training/losses/scalers/base_scaler.py [hunk 1]
```diff
@@ -185,7 +186,7 @@ def get_scaling(self) -> SCALER_DTYPE:
         scale_dims = tuple(x.value for x in self.scale_dims)
         return scale_dims, scalar_values
 
-    def get_callback_scaling_values(self, callback: AvailableCallbacks, **kwargs) -> SCALER_DTYPE:
+    def get_callback_scaling_values(self, callback: AvailableCallbacks, **kwargs) -> TENSOR_SPEC:
```
Comment (anaprietonem) [inline training/src/anemoi/training/losses/scalers/base_scaler.py]: how are the scalers expected to change across a training epoch or from training to validation?

Comment (sahahner) [reply training/src/anemoi/training/losses/scalers/base_scaler.py]: For the updating scalers, the value of the scalers can change from batch to batch.

**Comment by ssmmnn11** [comment]
in aifs-mono, we had forecaster (LightningModule)-> has self.loss (loss is a nn.Module), loss has buffers: "self.register_buffer("weights", area_weights, persistent=True)" so when you create the forecaster and move it to the GPU, all this is moved to the GPU

can you implement the same for the scalers / scale tensors? during init, everything required is registered as buffer, then every update can happen on device. remove all .to() from the code.

**Comment by sahahner** [comment]
Thank you Harrison for your work on this.
To me this is ready to be merged.

**Review by sahahner — APPROVED** [review]
Thank you Harrison for your work on this.
To me this is ready to be merged.

**Review by ssmmnn11 — CHANGES_REQUESTED** [review]
Very good. Can you do some test runs and check runtimes?

File: training/src/anemoi/training/losses/scaler_tensor.py [hunk 2]
```diff
@@ -633,11 +653,6 @@ def get_scaler(self, ndim: int, device: str | None = None) -> torch.Tensor:
             return complete_scaler.to(device)
```
Comment (ssmmnn11) [inline training/src/anemoi/training/losses/scaler_tensor.py:653]: if complete_scaler is None we would get torch.ones(1), which would then need to be moved on  device? Here, this would still happen in every call of the routine, is that correct? could you do either torch.ones(1, device=device) or torch.ones(1, device=some_reference_tensor)?

**Review by ssmmnn11 — CHANGES_REQUESTED** [review]
